### ISSUE TITLE
CA2000 False positive

### ISSUE THREAD

**Issue by amrmahdi** [dotnet/roslyn-analyzers#2201]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

[2.9.0-beta1.19158.2](https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.0-beta1.19158.2+85a05d0e)

#### Diagnostic ID
[CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope?view=vs-2017)

#### Repro steps

```cs
try
{
    using (var ms = new MemoryStream())
    {
        // do something with memory stream
    }
}
catch
{
}
```

#### Expected behavior

CA2000 not triggered.

#### Actual behavior

CA20000 triggered.
